tests/serialization/send.cpp

```diff
@@ -0,0 +1,57 @@
+#include <charmlite/charmlite.hpp>
+
+#include <iostream>
+#include <random>
+#include <tuple>
+#include <type_traits>
+#include <vector>
+
+struct invoker : cmk::chare<invoker, int>
+{
+    invoker(double arg0, std::vector<int> arg1)
```

Comment:
Can these be changed to `std::vector<int>&&`?

Reply:
Ohh absolutely. Let me do that! The way it works is that the message type and function arguments are related using `std::decay_t` so while the invoker will uphold the types on functions, we still end up with correct unwrapping.

Reply:
See the broadcast example where I take things by ref (it's about the same but extends the lifetime of the unwrapped variable until the end of the broadcasted function)

Reply:
Aah wait, you can't do rvalue_ref with it. The way it works is that the tuple parameters are directly sent as function parameters. This means that it can only be an lvalue or an lvalue reference. Is there an immediate need for rvalue reference? @jszaday